### ISSUE TITLE
Django >= 1.8 ; Deprecated OptParser usage ArgParse; Warnings on running runserver_plus

### ISSUE THREAD

**Issue by sudheesh001** [django-extensions/django-extensions#788]
Running `runserver_plus` with `Werkzeug` debugger throws the following warning with the latest Django

```
RemovedInDjango110Warning: OptionParser usage for Django management commands is deprecated, use ArgumentParser instead
  RemovedInDjango110Warning)
```

Looks like this is happening because of the usage of `*args` and `**options`,  See the changed in 1.8 which says https://docs.djangoproject.com/en/1.9/howto/custom-management-commands/#module-django.core.management
```
Changed in Django 1.8:
Before Django 1.8, management commands were based on the optparse module, and positional 
arguments were passed in *args while optional arguments were passed in **options. Now that 
management commands use argparse for argument parsing, all arguments are passed in **options 
by default, unless you name your positional arguments to args (compatibility mode). You are 
encouraged to exclusively use **options for new commands.
```

**Comment by nehemiahjacob** [comment]
+1

**Comment by thecardcheat** [comment]
Please see my #804.

**Comment by thecardcheat** [comment]
#804 should close this.

**Comment by timgraham** [comment]
`runscript` is still using `optparse`, and I'm not sure how to convert the `vararg_callback` part of it. Here's a start:
```diff
diff --git a/django_extensions/management/commands/runscript.py b/django_extensions/management/commands/runscript.py
index f7c01f0..a14b1a4 100644
--- a/django_extensions/management/commands/runscript.py
+++ b/django_extensions/management/commands/runscript.py
@@ -1,8 +1,8 @@
 # coding=utf-8
+import argparse
 import importlib
 import sys
 import traceback
-from optparse import make_option
 
 from django.apps import apps
 
@@ -11,39 +11,36 @@ from django_extensions.management.email_notifications import \
 from django_extensions.management.utils import signalcommand
 
 
-def vararg_callback(option, opt_str, opt_value, parser):
-    parser.rargs.insert(0, opt_value)
-    value = []
-    for arg in parser.rargs:
-        # stop on --foo like options
-        if arg[:2] == "--" and len(arg) > 2:
-            break
-        # stop on -a like options
-        if arg[:1] == "-":
-            break
-        value.append(arg)
+class VarArgCallback(argparse.Action):
 
-    del parser.rargs[:len(value)]
-    setattr(parser.values, option.dest, value)
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, values)
 
 
 class Command(EmailNotificationCommand):
-    option_list = EmailNotificationCommand.option_list + (
-        make_option('--fixtures', action='store_true', dest='infixtures', default=False,
-                    help='Only look in app.fixtures subdir'),
-        make_option('--noscripts', action='store_true', dest='noscripts', default=False,
-                    help='Look in app.scripts subdir'),
-        make_option('-s', '--silent', action='store_true', dest='silent', default=False,
-                    help='Run silently, do not show errors and tracebacks'),
-        make_option('--no-traceback', action='store_true', dest='no_traceback', default=False,
-                    help='Do not show tracebacks'),
-        make_option('--script-args', action='callback', callback=vararg_callback, type='string',
-                    help='Space-separated argument list to be passed to the scripts. Note that the '
-                         'same arguments will be passed to all named scripts.'),
-    )
     help = 'Runs a script in django context.'
     args = "script [script ...]"
 
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+        parser.add_argument(
+            '--fixtures', action='store_true', dest='infixtures', default=False,
+            help='Only look in app.fixtures subdir')
+        parser.add_argument(
+            '--noscripts', action='store_true', dest='noscripts', default=False,
+            help='Look in app.scripts subdir'
+        )
+        parser.add_argument(
+            '-s', '--silent', action='store_true', dest='silent', default=False,
+            help='Run silently, do not show errors and tracebacks')
+        parser.add_argument(
+            '--no-traceback', action='store_true', dest='no_traceback', default=False,
+            help='Do not show tracebacks')
+        parser.add_argument(
+            '--script-args', action=VarArgCallback, type=str,
+            help='Space-separated argument list to be passed to the scripts. Note that the '
+                 'same arguments will be passed to all named scripts.')
+
     @signalcommand
     def handle(self, *scripts, **options):
         NOTICE = self.style.SQL_TABLE
```
It's clearly incomplete but it passes the test so I guess good test coverage is missing.

If someone is able to finish this task, that will unblock me from dropping Django 1.6/1.7 support and adding Django 1.10 support (https://github.com/django-extensions/django-extensions/pull/913).

**Comment by trbs** [comment]
Doesn't the nargs='*' option more or less do what the old ```vararg_callback``` does ? https://docs.python.org/3/library/argparse.html#nargs

**Comment by trbs** [comment]
If there are issues left please open a new ticket for those :)

Thanks guys !

**Comment by benf-wspdigital** [comment]
> trbs closed this on 9 Aug 2016

What version of this code base contains the resolution?
